functional/pipeline.py

```diff
@@ -1724,3 +1732,64 @@ def _wrap(value):
         return Sequence(value)
     else:
         return value
+
+def extend(func=None, aslist=False, final=False):
```

Comment:
I think it would make sense to add a `name` argument here. For example, if you wanted extend something like `exec`, unless you want to override the builtin `exec` you wouldn't be able to call it like `seq().exec()`. To make things work, I think the default argument would need to be `None`, then immediately check if `name is None` then set it to `func.__name__`

Reply:
Another argument I think would make sense is `parallel`, to indicate whether the function is similar to `map` in that it is embarrassingly parallel.

Reply:
Good point. I'll add it.